### PR TITLE
Usman - Airport Challenge

### PR DESCRIPTION
**Pull request by UsmanJ**

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -1,2 +1,43 @@
...
+
+  def weather
+    [:sunny, :stormy].shuffle.first
+  end
+
+  def land_bad_weather
+    fail 'Can not take off in this weather' if weather = :stormy
+  end
+
+  def land(plane)
```
Comment (student-progress-tracker-dev) [inline lib/airport.rb]: Doesn't depend on instance state (UtilityFunction)

**Comment by tansaku** [comment]
Well done @UsmanJ; especially dealing with all the hound issues.

This is almost complete, just a couple of things:

*) public attr_accessor in plane.rb:5 is something you want to avoid.  As we talked about this week - avoid exposing the ability to directly manipulate internal state as part of ensuring encapsulation
*) no need for explicit return statements in ruby (hound should catch that)
*) avoid multiple expects in unit test it statements - use allow where appropriate for stubbing
*) pull utility functions like weather out into a module to help keep class responsibilities down
*) remove all commented code

Fix that up and I'll give you a checkered flag :-)

:construction:

**Comment by tansaku** [comment]
Hi @UsmanJ thanks for these updates.  We could give you a sticker, but see what you think about these other issues that we might also want to wrap up. Then tell me if you think you are ready for a sticker.

If you run rspec on this project you may see these messages:

```
WARNING: Using the `raise_error` matcher without providing a specific error or message risks false positives, since `raise_error` will match when Ruby raises a `NoMethodError`, `NameError` or `ArgumentError`, potentially allowing the expectation to pass without even executing the method you are intending to call. Instead consider providing a specific error class or message. This message can be supressed by setting: `RSpec::Expectations.configuration.warn_about_potential_false_positives = false`. Called from /Users/tansaku/Documents/Github/MakersAcademy/airport_challenge/spec/plane_spec.rb:13:in `block (2 levels) in <top (required)>'.
```

You've been splitting some of your lines in your spec files to avoid the 80 character limit, but it has actually changed the functionality of your code.  Can you see some other ways to address this issue?

You have created some dead code and I also see some others places where there is unecessary code hanging around plane_spec.rb#7 for example.  This is the kind of thing that hiring partners are going to frown about ...

I'm not too keen on the method names in the Plane class.  #can_land and #can_take_off seem like questions but they are actually commands to the Plane to change state - to make this really clean code you'll need to change them to reflect that.

plane.rb#21 I'd use not rather than ! here, and I might be tempted to clean out the instance variable with a private attr_reader.

* Please do update your README following the [contribution notes](https://github.com/makersacademy/airport-challenge/blob/master/CONTRIBUTING.md), i.e.

  * Make sure you have written your own README that briefly explains your approach to solving the challenge.
  * If your code isn't finished it's not ideal but acceptable as long as you explain in your README where you got to and how you would plan to finish the challenge.

* The above is a relatively straightforward thing to do that doesn't involve much programming - I'll often get it done while thinking about other problems in the back of my mind :-)  In particular what I'd really like to see is an irb transcript that shows me unequivocally how the system is intended to be used:

```sh
→ irb
2.2.3 :001 > airport = Airport.new
 => #<Airport:0x007fd2099dd540 @planes=[], @capacity=80> 
2.2.3 :002 > plane = Plane.new
ArgumentError: wrong number of arguments (0 for 1)
	from /Users/tansaku/Documents/Github/MakersAcademy/airport_challenge/lib/plane.rb:5:in `initialize'
	from (irb):2:in `new'
	from (irb):2
	from /Users/tansaku/.rvm/rubies/ruby-2.2.3/bin/irb:15:in `<main>'
2.2.3 :003 > plane = Plane.new :london
 => #<Plane:0x007fd2099a5c80 @flying=true, @destination=:london> 
2.2.3 :004 > airport.land plane
RuntimeError: Stormy weather
	from /Users/tansaku/Documents/Github/MakersAcademy/airport_challenge/lib/airport.rb:18:in `land'
	from (irb):4
	from /Users/tansaku/.rvm/rubies/ruby-2.2.3/bin/irb:15:in `<main>'
2.2.3 :005 > airport.land plane
RuntimeError: Stormy weather
	from /Users/tansaku/Documents/Github/MakersAcademy/airport_challenge/lib/airport.rb:18:in `land'
	from (irb):5
	from /Users/tansaku/.rvm/rubies/ruby-2.2.3/bin/irb:15:in `<main>'
2.2.3 :006 > airport.land plane
RuntimeError: Stormy weather
	from /Users/tansaku/Documents/Github/MakersAcademy/airport_challenge/lib/airport.rb:18:in `land'
	from (irb):6
	from /Users/tansaku/.rvm/rubies/ruby-2.2.3/bin/irb:15:in `<main>'
2.2.3 :007 > airport.land plane
RuntimeError: Stormy weather
	from /Users/tansaku/Documents/Github/MakersAcademy/airport_challenge/lib/airport.rb:18:in `land'
	from (irb):7
	from /Users/tansaku/.rvm/rubies/ruby-2.2.3/bin/irb:15:in `<main>'
2.2.3 :008 > airport.land plane
 => [#<Plane:0x007fd2099a5c80 @flying=true, @destination=:london>] 
```

* Some of your indenting is off in airport_spec.rb, and note that your single line tests of respond_to become redundant once you are actively testing the return values and side effects of those methods

* you make code like `if weather == :stormy` a bit more readable with some convenience methods, e.g. `if weather.stormy?`

Great effort so far.  I think you have the potential to do really well - shall we try and take your code to the next level?

File: lib/plane.rb [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+  end
+
+  def land
+    fail 'Can not land if it is not flying' if landed?
+    @flying = false
+    true
+  end
+
+  def landed?
+    not flying
```
Comment (UsmanJ) [inline lib/plane.rb]: Kept <code>not</code> after receiving Tansaku's feedback.

**Comment by tansaku** [comment]
cool - you've earned a sticker - the README still has some cruft related to the challenge, but that's easily fixed - well done

:checkered_flag:

**Pull request closed without merging**